prow/cmd/deck/tide.go

```diff
@@ -90,3 +95,64 @@ func (ta *tideAgent) update() error {
 	ta.pools = pools
 	return nil
 }
+
+func (ta *tideAgent) filterHidden(tideQueries []config.TideQuery, pools []tide.Pool) ([]config.TideQuery, []tide.Pool) {
+	if len(ta.hiddenRepos) == 0 {
+		return tideQueries, pools
+	}
+
+	var filteredTideQueries []config.TideQuery
+	for _, qc := range tideQueries {
+		includesHidden := false
+		// This will exclude the query even if a single
+		// repo in the query is included in hiddenRepos.
+		for _, repo := range qc.Repos {
+			if matches(repo, ta.hiddenRepos) {
+				includesHidden = true
+			}
+		}
+		if (includesHidden && ta.hiddenOnly) ||
+			(!includesHidden && !ta.hiddenOnly) {
+			filteredTideQueries = append(filteredTideQueries, qc)
+		}
+	}
+
+	var filteredPools []tide.Pool
+	for _, pool := range pools {
+		needsHide := matches(pool.Org+"/"+pool.Repo, ta.hiddenRepos)
+		if (needsHide && ta.hiddenOnly) ||
+			!needsHide && !ta.hiddenOnly {
+			filteredPools = append(filteredPools, pool)
+		}
+	}
+
+	return filteredTideQueries, filteredPools
+}
+
+// matches returns whether the provided repo intersects
+// with repos. repo and repos can either be actual repositories
+// (kubernetes/test-infra) or organizations (kubernetes).
+func matches(repo string, repos []string) bool {
```

Comment:
`repo` should never be just an "org". It should always be "org/repo" because it comes from the `Repos` field of `config.TideQuery`.

Reply:
We should validate this in `prow/config`.

Reply:
fixed